Added todo

Shtorkaa opened the pull request.

Shtorkaa committed:
Added todo

keksum666 commented on app/Http/Controllers/TaskController.php:
```diff
@@ -0,0 +1,31 @@
+class TaskController extends Controller
+{
+    public function addTask(Request $request){
+        DB::table('tasks')->insert([
```
Используем только модели, почитай полностью блок про модели

keksum666 commented on app/Http/Controllers/TaskController.php:
```diff
@@ -0,0 +1,31 @@
+
+class TaskController extends Controller
+{
+    public function addTask(Request $request){
```
Почитай про formRequest

keksum666 replied:
Давай без всякий постфиксом, просто create/show/delete/index

keksum666 commented on app/Models/task.php:
```diff
@@ -0,0 +1,10 @@
+
+use Illuminate\Database\Eloquent\Model;
+
+class task extends Model
```
Читаем правила наименования: https://laravel.demiart.ru/laravel-naming-conventions/

Используй fillable

keksum666 commented on routes/web.php:
```diff
@@ -2,14 +2,19 @@
-})->middleware(['auth', 'verified'])->name('dashboard');
+Route::get('/todo', [PageController::class, 'index'])->name('todo')->middleware('auth');
+
+Route::POST('/addTask', [TaskController::class, 'addTask'])->name('addTask')->middleware('auth');
```
POST маленькими буквами, почитай про роут, используй группы, добавь префикс

Shtorkaa committed:
Fixes

Added form requests, changed task model name and added fillable, reformat code, changed controllers to use models, grouped routes and changed methods

keksum666 commented on app/Http/Controllers/TaskController.php:
```diff
@@ -2,29 +2,39 @@
 
-    public function deleteTask(Request $request){
-        Task::destroy($request->id);
+    public function delete(DeleteTaskRequest $request)
```
Почитай про ModelBinding, тогда тебе не понадобится FormRequest

keksum666 commented on app/Http/Controllers/TaskController.php:
```diff
@@ -2,29 +2,39 @@
-            ->where('id', $request->id)
-            ->update(['title' => $request->title]);
-        return redirect()->back()->with('status' , 'Task updated');
+    public function edit(UpdateTaskRequest $request)
```
Примени тоже ModelBinging

keksum666 commented on app/Http/Controllers/TaskController.php:
```diff
@@ -2,29 +2,39 @@
+    {
+        $task = Task::find($request->validated()['id']);
+
+        $task->title = $request->validated()['title'];
```
Используй массовое присвоение

keksum666 commented on app/Http/Requests/DeleteTaskRequest.php:
```diff
@@ -0,0 +1,29 @@
+use Illuminate\Foundation\Http\FormRequest;
+use Auth;
+
+class DeleteTaskRequest extends FormRequest
```
Лишнее

keksum666 commented on app/Http/Requests/UpdateTaskRequest.php:
```diff
@@ -0,0 +1,30 @@
+    {
+        return [
+            'title' => 'required|max:255',
+            'id' => 'required',
```
Лишнее

keksum666 commented on app/Models/task.php:
```diff
@@ -6,5 +6,5 @@
 
 class task extends Model
```
Так и не прочитал про правила наименования + PSR

keksum666 replied:
Тут так и не поменял

keksum666 commented on database/migrations/2025_02_25_071137_create_tasks_table.php:
```diff
@@ -0,0 +1,28 @@
+    {
+        Schema::create('tasks', function (Blueprint $table) {
+            $table->id();
+            $table->string('title');
```
Тут привязка к пользователю должна быть, потому что пользователь свои задания должен видеть

Shtorkaa committed:
Fixes 2

added model binding for update, edit, added user_id for task migration

keksum666 commented on app/Http/Controllers/PageController.php:
```diff
@@ -0,0 +1,18 @@
+{
+    public function index()
+    {
+        $tasks = Task::where('user_id',Auth::user()->id)->get();
```
Вытягивыем через связи + пагинацию

keksum666 commented on app/Http/Controllers/TaskController.php:
```diff
@@ -0,0 +1,40 @@
+    {        
+        Task::create([
+            'title' => $request->validated()['title'],
+            'user_id' => Auth::user()->id,
```
Создавать нужно через связи

keksum666 replied:
Посмотри чт возвращает $request->validated(), можно просто его передавать в create, user_id не указываем, т.к мы создаем все через связи

keksum666 commented on app/Http/Controllers/TaskController.php:
```diff
@@ -0,0 +1,40 @@
+
+    public function edit(Request $request,  Task $task)
+    {
+        $task->title = $request->title;
```
Лучше использовать метод для обноволения который использует массовое заполнение

keksum666 commented on app/Models/task.php:
```diff
@@ -0,0 +1,10 @@
+
+class task extends Model
+{
+    protected $fillable = ['title','user_id'];
```
Давай делать многострочные массивы, то есть вот так:

`
[
   'title',
   'user_id',
];
`

+ по PSR всегда ставится завершающая запятая

keksum666 commented on database/migrations/2025_02_25_071137_create_tasks_table.php:
```diff
@@ -0,0 +1,31 @@
+    {
+        Schema::create('tasks', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('user_id');
```
Лучше и легче использовать foreignId

keksum666 commented on routes/web.php:
```diff
@@ -2,19 +2,26 @@
-})->middleware(['auth', 'verified'])->name('dashboard');
+Route::get('/todo', [PageController::class, 'index'])->name('todo')->middleware('auth');
+
+Route::middleware('auth')->group(function () {
```
name task можно повесить на группу, чотобы в каждом методе не использовать task.

+ добавь префикс task на саму группу

keksum666 commented on app/Http/Controllers/PageController.php:
```diff
@@ -0,0 +1,18 @@
+use Auth;
+class PageController extends Controller
+{
+    public function index()
```
Да и лучше вынеси этот метод в TaskController

Shtorkaa committed:
Added policy, minor fixes

Added policy for tasks, changed unsignedBigInt to foreignId in task migration, added prefix for task routes, moved geting task from page controller to task controller

Shtorkaa committed:
added pagination

keksum666 commented on app/Http/Controllers/TaskController.php:
```diff
@@ -0,0 +1,55 @@
+
+
+        $task->update([
+            'title' => $request->title,
```
Давай тоже это все делать через FormRequest

keksum666 commented on app/Http/Controllers/TaskController.php:
```diff
@@ -0,0 +1,55 @@
+    public function edit(Request $request, Task $task)
+    {
+        Gate::authorize('update', $task);
+
```
Пустая строка для чего?

keksum666 commented on routes/web.php:
```diff
@@ -2,19 +2,24 @@
+Route::get('/todo', [TaskController::class, 'index'])->name('todo')->middleware('auth');
+
+Route::middleware('auth')->name('task.')->prefix('task')->group(function () {
+    Route::post('/task', [TaskController::class, 'create'])->name('create');
```
У тебя из-за того что у тебя есть префикс будет: /task/task, убери везде task просто

Shtorkaa committed:
fixes
